src/globus_sdk/_testing/models.py

```diff
@@ -40,14 +59,44 @@ class RegisteredResponse:
     def __init__(
         self,
         *,
+        # path and service are glbous-sdk specific
+        # in `responses`, these are just `url`
         path: str,
-        service: str | None = None,
-        method: str = responses.GET,
+        service: (
+            Literal[
+                "auth",
+                "nexus",
+                "transfer",
+                "search",
+                "gcs",
+                "groups",
+                "timer",
+                "flows",
+                "compute",
+            ]
+            | None
+        ) = None,
+        # method derives from `responses.BaseResponse`
+        method: Literal["GET", "PUT", "POST", "PATCH", "HEAD", "DELETE"] = "GET",
```

Comment:
There are other methods, like `OPTIONS`. Would it makes sense to construct the literals from the `http.HTTPMethod` enum?

Also, the "derives from" phrasing isn't clear to me. Does that mean these literal values are the only ones supported by `responses.BaseResponse`, and should perfectly match it? Does that mean that `method` will be passed to a constructor that is originally defined in `responses.BaseResponse`?

Ultimately, these are passed as grouped keyword arguments to a single location, so it may be clearer to group these based on that final target, rather than noting the subclassing hierarchy.

Reply:
Let me pull the exact strings from `http.HTTPMethod` in case I missed others other than `OPTIONS`.

Dynamically building the Literal at runtime is a no-go. Type checkers and other analyzers won't be able to read the data because they won't execute the code.
We can add a test-case which asserts that the two are equal, which is what I will do.

Reply:
I'm not certain that I got exactly what you were trying to suggest here.
I've just implemented a change which I think resolves this, but feel free to un-resolve if you think there's more to discuss.